lib/galaxy/webapps/galaxy/controllers/async.py

```diff
@@ -84,12 +97,27 @@ def index(self, trans, tool_id=None, data_secret=None, **kwd):
             #
             # no data_id must be parameter submission
             #
+            GALAXY_TYPE = None
             if params.data_type:
                 GALAXY_TYPE = params.data_type
             elif params.galaxyFileFormat == 'wig':  # this is an undocumented legacy special case
                 GALAXY_TYPE = 'wig'
             else:
-                GALAXY_TYPE = params.GALAXY_TYPE or tool.outputs.values()[0].format
+                # Assume there is exactly one output
+                outputs_count = 0
+                for obj in tool.outputs.values():
+                    try:
+                        GALAXY_TYPE = params.GALAXY_TYPE or obj.format
+                        outputs_count += 1
+                        break
+                    except:
+                        # exclude outputs different from ToolOutput (e.g. collections) from the previous assumption
+                        continue
+                if outputs_count > 1:
+                    raise Exception( "Error: the tool should have just one output" )
```

Comment:
This exception will be thrown for the tools that do not respect the constraint of defining just one output in the xml schema (except for collections), as described in the comment ```Assume there is exactly one output```. Does it make sense for you?